### PR TITLE
Other refactored scripts on insert_atmp_t17

### PR DESCRIPTION
**Pull request by orgestbelba**

### CODE REVIEW

File: src/insert_atmp_t17/calc_grossinteres_mtd_refactored.py
Comment (dardanxh) [inline src/insert_atmp_t17/calc_grossinteres_mtd_refactored.py:1]: We can delete this file. We already have refactored it under `calc_gross_interest_mtd_refactored.py`.

File: src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        UPDATE "ABACUS_A5"."atmp_t17_dpd_credit_cards" AS t17
+        SET date_since_pd_ho2 = temp.date_since_pd_ho2,
+            dpd_ho2 = temp.dpd_ho2
+        FROM "ABACUS_A5"."temp_updated_rows" AS temp
+        WHERE t17.id_product = temp.id_product;
+        """
+
+        update_to_database(update_query)
+
+    def calculate_cc_dpd_ho2(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py:117]: This should be `run(self)` to comply with the `StructuredEtl`'s contract.

File: src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+from db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+    update_to_database,
+)
+from src.functions.func_date_since_pd_ho2_cc import func_date_since_pd_ho2_cc
+
+
+class CreditCardDPDHO2Calculator:
```
Comment (dardanxh) [inline src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py:12]: This should inherit from `StructuredEtl` class. Check example at `calc_gca_value_refactored.py`

File: src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    def __init__(self):
+        self.spark = get_spark_session()
+        self.postgres_properties = get_postgres_properties()
+        self.schema_name = "ABACUS_A5"
+
+    def read_data(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/calculate_cc_dpd_ho2_refactored.py:18]: This should be `read(self)` to comply with the `StructuredEtl`'s contract.

File: src/insert_atmp_t17/calculate_cc_dpd_ho_refactored.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+from db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+    update_to_database,
+)
+from src.functions.func_cust_materiality_level import func_cust_materiality_level
+
+
+class CreditCardDPDCalculator:
```
Comment (dardanxh) [inline src/insert_atmp_t17/calculate_cc_dpd_ho_refactored.py:12]: This should inherit from `StructuredEtl` class. Check example at `calc_gca_value_refactored.py`

File: src/insert_atmp_t17/calculate_cc_dpd_ho_refactored.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            SET date_since_pd_ho = temp.date_since_pd_ho,
+                dpd_ho = temp.dpd_ho,
+                materiality_level = temp.materiality_level
+            FROM "ABACUS_A5"."temp_updated_rows" AS temp
+            WHERE t17.id_product = temp.id_product;
+        """
+
+        update_to_database(update_query)
+
+    def calculate_cc_dpd_ho(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/calculate_cc_dpd_ho_refactored.py:172]: Override `StructuredEtl` functions.

File: src/insert_atmp_t17/check_customer_product_refactored.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+        # Insert new records
+        if df_inserts.count() > 0:
+            df_inserts.write.jdbc(
+                url=self.postgres_properties["url"],
+                table=f'"{schema_name}"."t02_customer_product"',
+                mode="append",
+                properties=self.postgres_properties,
+            )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/check_customer_product_refactored.py:132]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/check_customer_refactored.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    def persist(self, df_new_customers: DataFrame) -> None:
+        # Insert new customers into t01_customer if there are any
+        if df_new_customers.count() > 0:
+            df_new_customers.write.jdbc(
+                url=self.postgres_properties["url"],
+                table='"ABACUS_A5"."t01_customer"',
+                mode="append",
+                properties=self.postgres_properties,
+            )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/check_customer_refactored.py:54]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/check_product_type_refactored.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
+from pyspark.sql.functions import col, trim, lit
+from ..db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+)
+from ..core.structured_etl import StructuredEtl
```
Comment (dardanxh) [inline src/insert_atmp_t17/check_product_type_refactored.py:7]: Fix import. `core` package does not exist anymore.

File: src/insert_atmp_t17/check_product_type_refactored.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+        # Insert new product types into tc01_product_type
+        if df_new_product_types.count() > 0:
+            df_new_product_types.write.jdbc(
+                url=self.postgres_properties["url"],
+                table=f'"{schema_name}"."tc01_product_type"',
+                mode="append",
+                properties=self.postgres_properties,
+            )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/check_product_type_refactored.py:56]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/commit_atmp17_into_t17_refactored.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
+from pyspark.sql import functions as F
+from pyspark.sql.functions import col, lit
+from ..db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+)
+from ..core.structured_etl import StructuredEtl
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_atmp17_into_t17_refactored.py:8]: Fix import. `core` does not exist.

File: src/insert_atmp_t17/commit_atmp17_into_t17_refactored.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+        schema_name = "ABACUS_A5"
+
+        # Insert the data into t17_dpd_credit_cards
+        df_insert.write.jdbc(
+            url=self.postgres_properties["url"],
+            table=f'"{schema_name}"."t17_dpd_credit_cards"',
+            mode="append",
+            properties=self.postgres_properties,
+        )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_atmp17_into_t17_refactored.py:113]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/commit_atmp18_into_t18_refactored.py [hunk 12]
```diff
@@ -0,0 +1,71 @@
+from pyspark.sql import functions as F
+from pyspark.sql.functions import col
+from ..db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+)
+from ..core.structured_etl import StructuredEtl
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_atmp18_into_t18_refactored.py:8]: Fix import.

File: src/insert_atmp_t17/commit_atmp18_into_t18_refactored.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        schema_name = "ABACUS_A5"
+
+        # Insert the data into t18_cc_payment_schedule
+        df_insert.write.jdbc(
+            url=self.postgres_properties["url"],
+            table=f'"{schema_name}"."t18_cc_payment_schedule"',
+            mode="append",
+            properties=self.postgres_properties,
+        )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_atmp18_into_t18_refactored.py:60]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/commit_eatmpt17_into_t17_refactored.py [hunk 14]
```diff
@@ -0,0 +1,103 @@
+from pyspark.sql import functions as F
+from pyspark.sql.functions import col
+from ..db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+)
+from ..core.structured_etl import StructuredEtl
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_eatmpt17_into_t17_refactored.py:8]: Fix import.

File: src/insert_atmp_t17/commit_eatmpt17_into_t17_refactored.py [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+        schema_name = "ABACUS_A5"
+
+        # Insert the data into t17_dpd_credit_cards
+        df_insert.write.jdbc(
+            url=self.postgres_properties["url"],
+            table=f'"{schema_name}"."t17_dpd_credit_cards"',
+            mode="append",
+            properties=self.postgres_properties,
+        )
+
```
Comment (dardanxh) [inline src/insert_atmp_t17/commit_eatmpt17_into_t17_refactored.py:92]: Introduce function `run(self)` and bring there the functionality from the main below.

File: src/insert_atmp_t17/check_customer_refactored.py
Comment (dardanxh) [inline src/insert_atmp_t17/check_customer_refactored.py:1]: Delete this file. We already have refactored it.

File: src/insert_atmp_t17/isexistingcustomer_refactored.py
Comment (dardanxh) [inline src/insert_atmp_t17/isexistingcustomer_refactored.py:1]: This function does not require refactoring, as it is not a data processing script. We can revert it to its previous state.

File: src/insert_atmp_t17/load_prev_atmp_t17_refactored.py [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+from pyspark.sql.functions import col, lit
+from db_connection_util import (
+    get_spark_session,
+    get_postgres_properties,
+    read_data_from_postgres,
+    update_to_database,
+)
+
+
+class PreviousATMPUpdater:
```
Comment (dardanxh) [inline src/insert_atmp_t17/load_prev_atmp_t17_refactored.py:11]: Inherit from `StructuredEtl` class

File: src/insert_atmp_t17/load_prev_atmp_t17_refactored.py [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+    def __init__(self):
+        self.spark = get_spark_session()
+        self.postgres_properties = get_postgres_properties()
+        self.schema_name = "ABACUS_A5"
+
+    def read_data(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/load_prev_atmp_t17_refactored.py:17]: Follow `StructuredEtl` function namings.

File: src/insert_atmp_t17/load_prev_atmp_t17_refactored.py [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+        update_to_database(update_query)
+
+        # Update p_working_day for all records
+        update_working_day_query = f"""
+            UPDATE "{self.schema_name}"."atmp_t17_dpd_credit_cards"
+            SET p_working_day = '{prev_working_day}';
+        """
+        update_to_database(update_working_day_query)
+
+    def load_previous_atmp(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/load_prev_atmp_t17_refactored.py:126]: Rename to `run(self)` to comply with `StructuredEtl` function namings.

File: src/insert_atmp_t17/update_is_joint_refactored.py [hunk 19]
```diff
@@ -0,0 +1,83 @@
+from pyspark.sql import functions as F
+from pyspark.sql.functions import col, lit, when
+from db_connection_util import get_spark_session, get_postgres_properties
+
+
+class JointStatusUpdater:
```
Comment (dardanxh) [inline src/insert_atmp_t17/update_is_joint_refactored.py:6]: Inherit from `StructuredEtl`

File: src/insert_atmp_t17/update_is_joint_refactored.py [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+    def __init__(self):
+        self.spark = get_spark_session()
+        self.postgres_properties = get_postgres_properties()
+
+    def read_data(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/update_is_joint_refactored.py:11]: Follow `StructuredEtl` function namings.

File: src/insert_atmp_t17/update_is_joint_refactored.py [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+    def persist(self, df_updated):
+        # Write back to the database
+        df_updated.write.jdbc(
+            table="atmp_t17_dpd_credit_cards",
+            properties=self.postgres_properties,
+            mode="overwrite",  # Use 'overwrite' to update existing records
+        )
+
+    def update_joint_status(self):
```
Comment (dardanxh) [inline src/insert_atmp_t17/update_is_joint_refactored.py:71]: Rename to `run(self)` to comply with `StructuredEtl` function namings.

**Review by dardanxh — CHANGES_REQUESTED** [review]
The following changes need to be applied consistently across all scripts:

- Run `poetry run pre-commit run --all-files` in your terminal to perform code formatting.
- Ensure that packages are imported correctly.
- Ensure that each refactored script inherits from the StructuredEtl class.
- Ensure that every refactored script properly overrides the functions from the `StructuredEtl` class (e.g., `read()`, `transform()`, `persist()`, `run()`).
- Ensure that every function has a specified return data type in its function header.

**Review by dardanxh — APPROVED** [review]
Great job Denis 🚀

**Review by ergingorishti — APPROVED** [review]

**Pull request merged**